indexd/src/quic/download.rs

```diff
@@ -170,9 +170,38 @@ pub struct DownloaderInner {
     account_key: PrivateKey,
     host_client: Client,
     app_client: AppClient,
+    location: Mutex<Option<GeoLocation>>,
 }
 
 impl DownloaderInner {
+    async fn set_location(&self, location: Option<GeoLocation>) -> Result<(), app_client::Error> {
```

Comment:
I wouldn't bother with this. Either you create the `Downloader` with a location or not. If you IP address changed, your downloads got interrupted and you'll have to create a new one anyway.